src/main/resources/Sessions/Sessions.drl

```diff
@@ -38,11 +38,44 @@ rule "Session does not have a judge 4 weeks or less before start"
         Year($year: value)
         Month($month: value)
         Day($day: value)
-        $s : Session(judgeId == null && DateTimeUtils.between(start, $year, $month, $day, 0, 28))
+        $s : Session(judgeId == null && DateTimeUtils.between(start, $year, $month, $day, 15, 29))
     then
         insertLogical(new Problem(ProblemTypes.Resource_missing,
                                   ProblemSeverities.Warning,
-                                  "Session does not have a judge 4 weeks or less before start",
+                                  String.format("Session of type %s on %s does not yet have a judge assigned",
+                                                                    $s.getSessionType(),
+                                                                    DateTimeUtils.humanizeDate($s.getStart())),
+                                  new ProblemReference($s)));
+end
+
+rule "Session does not have a judge 2 weeks or less before start"
```

Comment:
Consider single rules with if for severity to allow parameters by configuration later on.